includes/site-health/audit-autoloaded-options/helper.php

```diff
@@ -197,28 +198,35 @@ function perflab_aao_get_autoloaded_options_table(): string {
  *
  * @since 3.0.0
  *
- * @global wpdb $wpdb WordPress database abstraction object.
- *
  * @return string HTML formatted table.
  */
 function perflab_aao_get_disabled_autoloaded_options_table(): string {
-	global $wpdb;
-
 	$disabled_options = get_option( 'perflab_aao_disabled_options', array() );
 
-	if ( empty( $disabled_options ) ) {
+	if ( ! is_array( $disabled_options ) ) {
 		return '';
 	}
 
-	$disabled_options_summary = $wpdb->get_results(
-		$wpdb->prepare(
-			sprintf(
-				"SELECT option_name, LENGTH(option_value) AS option_value_length FROM $wpdb->options WHERE option_name IN (%s) ORDER BY option_value_length DESC",
-				implode( ',', array_fill( 0, count( $disabled_options ), '%s' ) )
-			),
-			$disabled_options
-		)
-	);
+	$disabled_options_summary = array();
+	wp_prime_option_caches( $disabled_options );
+
+	foreach ( $disabled_options as $option_name ) {
+		if ( ! is_string( $option_name ) ) {
+			continue;
+		}
+		$option_value = get_option( $option_name );
+
+		if ( false !== $option_value ) {
+			$option_length                            = strlen( maybe_serialize( $option_value ) );
+			$disabled_options_summary[ $option_name ] = $option_length;
+		}
+	}
+
+	if ( count( $disabled_options_summary ) === 0 ) {
+		return '';
+	}
```

Comment:
I realized that with the refactor here, it is possible for `$disabled_options_summary` to be an empty array at this point, if none of the options are in the database anymore. This is quite unlikely, but to account for this I've added e6f8650889d585f28e29972e0214aa540764b7ae.